src/libNode/MicroBlockProcessing.cpp

```diff
@@ -1560,10 +1583,18 @@ bool Node::MicroBlockValidator(const vector<unsigned char>& microblock,
         return true;
     }
 
-    LOG_MARKER();
+    MicroBlock microBlock;
+
+    if (!Messenger::GetNodeMicroBlockAnnouncement(
+            message, offset, consensusID, blockHash, leaderID, leaderKey,
+            microBlock, messageToCosign))
+    {
+        LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+                  "Messenger::GetNodeMicroBlockAnnouncement failed.");
+        return false;
+    }
 
-    // [TODO] To put in the logic
-    m_microblock = make_shared<MicroBlock>(MicroBlock(microblock, 0));
+    m_microblock.reset(new MicroBlock(microBlock));
```

Comment:
If we wrap the `microBlock` in a unique_ptr, we could directly assign it to `m_microblock`  with the same effect but better efficiency.

Reply:
There're similar patterns in other locations in this PR. A large object is defined first, and copied over to a dynamically allocated object.

```cpp
LargeObject obj;

if (...)
{
    // do something with obj
}

m_obj_unique_ptr.reset(new LargeObject(object)); // an uncessary copy will happen if LargeObject doesn't have a move constructor
```